lib/puma/reactor.rb

```diff
@@ -39,7 +39,8 @@ def run_internal
 
         if ready and reads = ready[0]
           reads.each do |c|
-            if c == @ready
+            client = sockets.detect{|socket| socket.to_io == c }
```

Comment:
This line really hurts the performance of this acceptor loop because it means searching through all the sockets again on each iteration. In order to not penalize other platforms, I'd prefer to put this and the change on line 28 behind methods that are defined differently on maglev than other platforms. On other platforms, the methods would be noops that just return the value given. On maglev, the transformations would be applied.